lib/polars/catalog/api_client.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'json'
+require 'uri'
+
+module Polars
+  module Catalog
+    class Unity
+      class ApiClient
+        def initialize(workspace_url, bearer_token)
+          @base_uri = URI.parse(workspace_url)
+          @bearer_token = bearer_token
+        end
+
+        def get(path, params = {})
+          uri = @base_uri.dup
+          uri.path = path
+          uri.query = URI.encode_www_form(params) unless params.empty?
+
+          request = Net::HTTP::Get.new(uri)
+          perform_request(request)
+        end
+
+        def post(path, body)
+          uri = @base_uri.dup
+          uri.path = path
+
+          request = Net::HTTP::Post.new(uri)
+          request.body = body.to_json
+          perform_request(request)
+        end
```

Comment:
Because this is read only I never actually make a post request. Should I remove until its needed?